parsl/executors/high_throughput/executor.py

```diff
@@ -479,38 +506,60 @@ def scale_out(self, blocks=1):
         r = []
         for i in range(blocks):
             if self.provider:
-                block = self.provider.submit(self.launch_cmd, 1, 1)
-                logger.debug("Launched block {}:{}".format(i, block))
-                if not block:
+                external_block_id = len(self.blocks)
+                launch_cmd = self.launch_cmd.format(block_id=external_block_id)
+                internal_block = self.provider.submit(launch_cmd, 1, 1)
+                logger.debug("Launched block {}->{}".format(external_block_id, internal_block))
+                if not internal_block:
                     raise(ScalingFailed(self.provider.label,
                                         "Attempts to provision nodes via provider has failed"))
-                self.blocks.extend([block])
+                r.extend([str(external_block_id)])
+                self.blocks[str(external_block_id)] = internal_block
             else:
                 logger.error("No execution provider available")
                 r = None
         return r
 
-    def scale_in(self, blocks):
+    def scale_in(self, blocks, block_ids=[]):
```

Comment:
What is the expected way in which a user of this would acquire a block IDs to pass in here? Something that knows how to look into the guts of htex?

Reply:
The mechanism is to call `executor.connected_workers` which returns something like this : 
```python
[{'manager': 'cd17f5ff5268', 'block_id': '0', 'worker_count': 4, 'tasks': 0, 'active': True}] ```

This would allow the strategy component to identify block to manager to worker mappings and determine what capacity is available/in-use.